psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,86 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import Optional, Type
+
+from .. import postgres
+from ..abc import AdaptContext
+from ..adapt import Dumper, Loader
+from ..pq import Format
+from .._typeinfo import TypeInfo
+
+
+try:
+    import shapely.wkb as wkb
+    from shapely.geometry.base import BaseGeometry
+
+except ImportError:
+    raise ImportError(
+        "The module psycopg.types.geometry requires the package 'Shapely'"
+        " to be installed"
+    )
+
+
+class GeometryBinaryLoader(Loader):
+    format = Format.BINARY
+
+    def load(self, data: bytes) -> "BaseGeometry":
+        return wkb.loads(data)
+
+
+class GeometryLoader(Loader):
+    format = Format.TEXT
+
+    def load(self, data: bytes) -> "BaseGeometry":
+        # it's a hex string in binary
+        return wkb.loads(data.decode(), hex=True)
+
+
+class GeometryBinaryDumper(Dumper):
+    format = Format.BINARY
+
+    def dump(self, obj: "BaseGeometry") -> bytes:
+        return wkb.dumps(obj).encode()  # type: ignore
```

Comment:
I don't think you want `encode()` here. AFAICS, wkb.dumps() dumps already bytes.
```python
>>> wkb.dumps(point)
b'\x01\x01\x00\x00\x00333333\xf3?333333\x0b@'
```

Reply:
Right, wkb means literally _Well Known Binary_ :-)
Checking this I found out that Shapely already returns str or bytes depending on the `hex` parameter

Reply:
It's too bad, anyway, that the hex string is not accepted as bytes: it would be more efficient to parse. I also see a lot of ctypes: [for experience](https://www.varrazzo.com/blog/2020/05/19/a-trip-into-optimisation/) that's a great loss of performance.